### PR TITLE
Add DefaultReplacesNull qualifier example.

### PR DESCRIPTION
**Pull request by NightlyNexus**
A simpler example of a JsonQualifier with elements. Feel free to close if you had something else in mind, of course.

### CODE REVIEW

File: examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            break;
+          }
+        }
+        if (delegateAnnotations == null) {
+          return null;
+        }
+        JsonAdapter<Object> delegate = moshi.adapter(type, delegateAnnotations);
+        Object defaultIfNull;
+        try {
+          defaultIfNull = delegate.fromJson(defaultJson);
```
[the remark this answers is not in the document]
Comment (NightlyNexus) [reply examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java]: I'm having a tough time thinking of how to break this. Do you have a hint of how to make this corner case?

File: examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    @Override public Object fromJson(JsonReader reader) throws IOException {
+      Object result = delegate.fromJson(reader);
+      if (result == null) {
+        result = defaultIfNull;
+      }
+      return result;
+    }
+
+    @Override public void toJson(JsonWriter writer, @Nullable Object value) throws IOException {
+      if (value == null) {
```
[the remark this answers is not in the document]
Comment (NightlyNexus) [reply examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java]: oooh, yeah, that makes more sense.

File: examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import java.util.Set;
+import javax.annotation.Nullable;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+final class CustomQualifierWithElement {
+  @Retention(RUNTIME)
+  @JsonQualifier
+  public @interface DefaultReplacesNull {
+    String json();
```
[the remark this answers is not in the document]
Comment (NightlyNexus) [reply examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java]: true! very rare to be mutable in practice, i'd think, but worth stating.

[the remark this answers is not in the document]
Comment (NightlyNexus) [reply examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java]: ha, just realized you meant the deserialized value, not the annotation implementation.

File: examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (swankjesse) [reply examples/src/main/java/com/squareup/moshi/recipes/CustomQualifierWithElement.java]: Umm, maybe something like this?

```
  class Person {
    String name;

    @DefaultReplacesNull(json = "{\"name\":\"Just Bieber\"}")
    Person favorite_artist;
  }
```

Not sure if it really matters, but reentrancy can be surprising.

**Comment by NightlyNexus** [comment]
got busy and forgot about this.

i'd ideally like to come up with an example inspired from something we or somebody we know has had to implement before. closing in favor of #479.

**Pull request closed without merging**